contracts/upgradeable_contracts/erc20_to_native/HomeBridgeErcToNative.sol

```diff
@@ -84,10 +91,31 @@ contract HomeBridgeErcToNative is EternalStorage, BasicBridge, BasicHomeBridge,
         setTotalExecutedPerDay(getCurrentDay(), totalExecutedPerDay(getCurrentDay()).add(_value));
         IBlockReward blockReward = blockRewardContract();
         require(blockReward != address(0));
-        blockReward.addExtraReceiver(_value, _recipient);
+        uint256 valueToMint = _value;
+
+        address feeManager = feeManagerContract();
+        if (feeManager != address(0)) {
+            uint256 fee = calculateFee(valueToMint, false, feeManager);
+            distributeFeeFromAffirmation(fee, feeManager);
```

Comment:
Seems I found a potential bug: what if we have 3 validators and 1 gwei to distribute as a fee. In this case every validator will get 333333333 wei. The sum these values is 999999999 wei which is less than 1 gwei. Does it mean that the rest (1 wei) should be returned back to the user? It is applicable to transfer in both directions.
We will discuss this internally and I provide a suggestion. If you have any ideas please let me know.

Reply:
OK. We discussed this issue and the decision is to add the rest of fee to a validator chosen on pseudo-random manner. So, if a fee is 1 gwei, two validators will get per 333333333 and one will get 333333334.
In order to determine the validator-recipient for this value I suggest to use the block hash:
```solidity
    function random() public view returns(uint256) {
        return uint256(blockhash(block.number-1)) % validatorCount();
    }
```

Reply:
Thanks for suggestions. Fixed here https://github.com/poanetwork/poa-bridge-contracts/pull/132/commits/71870e0076fb801699c5c337bf52f221a214c83a

Reply:
great!